rust_src/src/eval.rs

```diff
@@ -565,7 +569,7 @@ pub fn eval(form: LispObject, lexical: LispObject) -> LispObject {
 #[no_mangle]
 pub extern "C" fn apply1(mut func: LispObject, arg: LispObject) -> LispObject {
     if arg == Qnil {
-        unsafe { Ffuncall(1, &mut func) }
+        Ffuncall(1, &mut func)
```

Comment:
can't we use `funcall` itself here ?

Reply:
see `eval.c:1912`. `Ffuncall` is the C name of the lisp function `funcall`, and it's not just a simple wrapper around a single C function. Instead it handles the eval depth check, backtraces, GC, interactive debugging, error handling, the `specpdl_ptr`, and finally it delegates to any of several other function depending on what you're calling (`funcall_subr`, `funcall_lambda`, and `Fautoload_do_load`). Eventually someone will port it to Rust and the `Ffuncall` name will go away.

Reply:
Yeah, I missed that.

Reply:
Good catch @shanavas786